addons/payment_stripe_sca/models/payment.py

```diff
@@ -54,6 +54,12 @@ def _stripe_request(self, url, data=False, method="POST"):
             resp.raise_for_status()
         except:
             _logger.error(resp.text)
+            try:
+                error = resp.json()
+            except ValueError:  # .json() raise ValueError if the response body does not contain valid json. So we fallback on original Exception
+                pass
+            else:
+                raise Exception(error.get('error', {}).get('message'))
```

Comment:
Looks overly complicated, why not something like

```python
try:
    resp.raise_for_status()
except Exception as e:
    _logger.error(resp.text)
    try:
        error = resp.json()
    except ValueError:
        raise e
    else:
        raise Exception(error.get('error', {}).get('message'))
```

Reply:
Yes i think richard is right; otherwise the raise at line 63 becomes ambiguous imho